Controller/Admin/Transaction/TransactionTabPostProcessing.php

```diff
@@ -157,68 +239,103 @@ private function _processRequest(array $aRequestParameters)
     /**
      * Returns an array of processing actions to be displayed below the amount input field.
      *
+     * @throws Exception in case the payment method type is not found
+     *
      * @return array
      *
-     * @since 1.0.0
+     * @since 1.0.1
      */
-    protected function _getPostProcessingActions(): array
+    protected function _getPostProcessingActions()
     {
-        // TODO return only supported actions for the payment
-        return [
-            'cancel' => [
-                'title' => Helper::translate('cancel'),
-                'callback' => '_onCancel',
-            ],
-            'capture' => [
-                'title' => Helper::translate('pay'),
-                'callback' => '_onCapture',
-            ],
-            'refund' => [
-                'title' => Helper::translate('refund'),
-                'callback' => '_onRefund',
-            ],
-        ];
+        $sPaymentId = $this->oTransaction->getPaymentType();
+        $oPayment = PaymentMethodHelper::getPaymentById($sPaymentId);
+
+        // Need to create a transaction object with the ID of the currently selected one to get
+        // the available post-processing operations from the Payment SDK
+        $oPaymentMethod = PaymentMethodFactory::create($sPaymentId);
+        $oTransaction = $oPaymentMethod->getTransaction();
+        $sParentTransactionId = $this->oTransaction->wdoxidee_ordertransactions__transactionid->value;
+        $oTransaction->setParentTransactionId($sParentTransactionId);
+
+        $oBackendService = new BackendService($oPaymentMethod->getConfig($oPayment), $this->_oLogger);
+        $aPossibleOperations = $oBackendService->retrieveBackendOperations($oTransaction, true);
+
+        if ($aPossibleOperations !== false && count($aPossibleOperations > 0)) {
+            return $this->_getTranslatedPostProcessingActions($aPossibleOperations, $oPaymentMethod);
```

Comment:
In my opinion, an **early exit** is used to handle an edge case, which is why I would turn it around in this case by checking for `$aPossibleOperations === false && count($aPossibleOperations <= 0)`, thus returning `[]`.

Reply:
Good point 👍